src/repl.ts

```diff
@@ -135,16 +146,45 @@ export function createRepl(options: CreateReplOptions = {}) {
           _console.error(error);
         }
       } else {
-        err = error;
+        err = error as Error;
+      }
+    }
+
+    try {
+      const { awaitPromise, result: evalResult } = evalCodeInternal(
+        code,
+        server.useGlobal ? undefined : context
+      );
+
+      if (awaitPromise) {
+        return (async () => {
+          try {
+            result = await evalResult;
+          } catch (promiseError) {
+            handleError(promiseError);
+          }
+
+          callback(err, result);
+        })();
       }
+
+      result = evalResult;
+    } catch (error) {
+      handleError(error);
     }
 
     return callback(err, result);
   }
 
-  function start(code?: string) {
+  function start(code?: string, optionsOverride?: ReplOptions) {
```

Comment:
Is `optionsOverride` only used by our tests?  If so, can we move it to a new `@internal` method to avoid affecting the public API surface?

I can see it being useful for API consumers, but it may be best to tackle in a separate PR.

Reply:
Yes, currently only used in tests, however I really find it highly useful for API consumers, as well as allowing accessing the repl server.